### PR TITLE
feat(workflow): Track Rule and AlertRule activity

### PR DESCRIPTION
**Pull request by iProgramStuff**
We're going to be displaying the user who created the rule in the alert listing. To accomplish this, we now log who created a rule and will fetch this data in the serializer.

Long-term we may want to display other activity somewhere so I've added some activity logging on create, update, and delete as well.

### CODE REVIEW

File: src/sentry/api/endpoints/project_rules.py [hunk 1]
```diff
@@ -52,7 +52,9 @@ def post(self, request, project):
             }}
 
         """
-        serializer = RuleSerializer(context={"project": project}, data=request.data)
+        serializer = RuleSerializer(
+            context={"project": project, "user": request.user}, data=request.data
```
Comment (wedamija) [inline src/sentry/api/endpoints/project_rules.py:56]: Why do we need to pass `user` to this serializer?

**Review by wedamija — COMMENTED** [review]
A little feedback, waiting on tests for the final pass

File: src/sentry/migrations/0094_ruleactivity.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    atomic = True
+
+
+    dependencies = [
+        ('sentry', '0093_make_identity_user_id_textfield'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='RuleActivity',
```
Comment (markstory) [inline src/sentry/migrations/0094_ruleactivity.py:38]: Does this new model need to be added to deletion code?

Comment (iProgramStuff) [reply src/sentry/migrations/0094_ruleactivity.py]: I don't think so. I could register it in deletions/__init__.py but I'd likely have to create a custom deletion task to ensure there are no active rules associated with an activity item. I'm thinking it's best to just ensure these are cleaned up when rules are deleted.

**Review by wedamija — APPROVED** [review]

**Pull request merged**